### PR TITLE
Support for different backpack window capacities

### PR DESCRIPTION
**Pull request by SoundSSGood**
This is what the infinity backpack looks like
infinite
64 items limit.
64_lim
30 items limit.
30_lim

### CODE REVIEW

**Comment by DjWarmonger** [comment]
Can you resize the window to make use of higher resolutions? Like, 10 x 10 and 12 x 12 variants - if needed.

**Comment by IvanSavenko** [comment]
Frankly filters / sorting would be much more useful than larger window. 
(not insisting on either option though)

**Comment by SoundSSGood** [comment]
> Can you resize the window to make use of higher resolutions? Like, 10 x 10 and 12 x 12 variants - if needed.

Yes, it's easy. If necessary, I can make the API for that

File: client/windows/CHeroBackpackWindow.cpp [hunk 1]
```diff
@@ -24,22 +24,20 @@ CHeroBackpackWindow::CHeroBackpackWindow(const CGHeroInstance * hero)
...
 	pos.w = stretchedBackground->pos.w;
 	pos.h = stretchedBackground->pos.h;
 	center();
-
-
-	arts = std::make_shared<CArtifactsOfHeroBackpack>(/*Point(-100, -170)*/Point(10, 10));
-	arts->setHero(hero);
-	addSet(arts);
-
+	std::iter_swap(GH.createdObj.front()->children.begin(), GH.createdObj.front()->children.begin() + 2);
```
Comment (IvanSavenko) [inline client/windows/CHeroBackpackWindow.cpp:35]: Why is this needed?

Comment (SoundSSGood) [reply client/windows/CHeroBackpackWindow.cpp:35]: First the CArtifactsOfHeroBackpack and CButton are created, then CFilledTexture. CFilledTexture is at the beginning of GH.createdObj.front()->children and will be displayed on top.

Comment (IvanSavenko) [reply client/windows/CHeroBackpackWindow.cpp:35]: Why can't we swap creation order and create background first?
If you don't know window size you can update it later, e.g:
```cpp
stretchedBackground = std::make_shared<CFilledTexture>( ... )
(create rest of widgets)
stretchedBackground->pos = actualPos
```

Comment (SoundSSGood) [reply client/windows/CHeroBackpackWindow.cpp:35]: Right. this approach works

**Review by IvanSavenko — APPROVED** [review]
